src/lib/components/MainPanel.js

```diff
@@ -66,18 +61,12 @@ export default class MainPanel extends React.Component {
     scheduleSnoozedTab(time);
   }
 
-  handleTimeSelect(value) {
-    this.setState({ dateChoice: value });
-  }
-
   closeTimeSelect() {
     this.setState({ datepickerActive: false });
   }
 
-  confirmTimeSelect() {
-    const { dateChoice } = this.state;
+  confirmTimeSelect(dateChoice) {
     const { scheduleSnoozedTab } = this.props;
-
     if (!dateChoice) { return; }
```

Comment:
So, I was able to set a time in the past by going to tomorrow, picking 9am, then going back to today.  I think we should probably check for that somewhere before here, but checking that the date/time is valid in `confirmTimeSelect` also seems like a reasonable thing to do.  😉